Assets/RedBlueGames/BulkRename/Editor/BulkRenamerWindow.cs

```diff
@@ -23,712 +23,755 @@
+			}
+
+			return names;
+		}
+
+		private void RenameAssets ()
+		{
+			// Record all the objects to undo stack, though this unfortunately doesn't capture Asset renames
+			Undo.RecordObjects (this.ObjectsToRename.ToArray (), "Bulk Rename");
+
+			var names = this.GetNamesFromObjectsToRename ();
+			var newNames = this.bulkRenamer.GetRenamePreviews (names);
+
+			for (int i = 0; i < newNames.Count; ++i) {
+				var infoString = string.Format (
+					                 "Renaming asset {0} of {1}",
+					                 i,
+					                 newNames.Count);
+
+				EditorUtility.DisplayProgressBar (
+					"Renaming Assets...",
+					infoString,
+					i / (float)newNames.Count);
+
+				this.RenameObject (this.ObjectsToRename [i], newNames [i].NewName);
+			}
+			RenameSprites (allSpritesWithNewNames);
+
+			EditorUtility.ClearProgressBar ();
+			allSpritesWithNewNames.Clear ();
+
+		}
+
+		private void RenameObject (UnityEngine.Object obj, string newName)
+		{
+			if (AssetDatabase.Contains (obj)) {
+				this.RenameAsset (obj, newName);
+			} else {
+				this.RenameGameObject (obj, newName);
+			}
+		}
+
+		private void RenameGameObject (UnityEngine.Object gameObject, string newName)
+		{
+			gameObject.name = newName;
+		}
+
+		public 	static string ReplaceSpritePrefixInMetafile (string metafileText, string prefixToReplace, string newPrefix)
+		{			
+			string fileIDPattern = prefixToReplace;
+			var fileIDRegex = new System.Text.RegularExpressions.Regex (fileIDPattern);
+			string replacementText = newPrefix;
+			return fileIDRegex.Replace (metafileText, replacementText);
+		}
+
+		void RenameSprites (Dictionary<Object,string> allSpritesWithNewNames)
+		{
+			string actualAssetName;
+			string newAssetName;
+			string path;
+			HashSet<string> assetsToReimport = new HashSet<string> ();
+
+			foreach (var asset in allSpritesWithNewNames) {
+				actualAssetName = asset.Key.name;
+				newAssetName = asset.Value;
+				path = AssetDatabase.GetAssetPath (asset.Key);
+				TextureImporter parentAsset = TextureImporter.GetAtPath (path)as TextureImporter;
+				#region Changing name in metafiles
+				PropertyInfo cachedInspectorModeInfo = typeof(SerializedObject).GetProperty ("inspectorMode", BindingFlags.NonPublic | BindingFlags.Instance);
+				SerializedObject so = new UnityEditor.SerializedObject (asset.Key);
+				cachedInspectorModeInfo.SetValue (so, InspectorMode.Debug, null);
+				PropertyInfo inspectorModeInfo = typeof(SerializedObject).GetProperty ("inspectorMode", BindingFlags.NonPublic | BindingFlags.Instance);
+				SerializedProperty sp = so.FindProperty ("m_LocalIdentfierInFile");
+				string oldNameMeta = sp.intValue + ": " + actualAssetName;
+				string newNameMeta = sp.intValue + ": " + asset.Value;
+				string metaFile = System.IO.File.ReadAllText (path + ".meta");
+				string ammendedMetaFile = ReplaceSpritePrefixInMetafile (metaFile, actualAssetName, newAssetName);
+				System.IO.File.WriteAllText (path + ".meta", ammendedMetaFile);
```

Comment:
Actually now that I see the part about AssetsToReimport you could basically group all that into this, too.